src/parquet/column/reader.cc

```diff
@@ -109,24 +97,30 @@ bool TypedColumnReader<TYPE>::ReadNewPage() {
       // the page size to determine the number of bytes in the encoded data.
       size_t data_size = page->size();
 
-      max_definition_level_ = descr_->max_definition_level();
-
-      // Read definition levels.
-      if (max_definition_level_ > 0) {
-        // Temporary hack until schema resolution implemented
-
-        size_t def_levels_bytes = InitializeLevelDecoder(buffer,
-            max_definition_level_, definition_level_decoder_);
-
+      int16_t max_definition_level = descr_->max_definition_level();
+      int16_t max_repetition_level = descr_->max_repetition_level();
+      //Data page Layout: Repetition Levels - Definition Levels - encoded values.
+      //Levels are encoded as rle or bit-packed.
+      //Init repetition levels
+      if (max_repetition_level > 0) {
+        size_t rep_levels_bytes = repetition_level_decoder_.Init(
+            page->repetition_level_encoding(),
+            max_repetition_level, num_buffered_values_, buffer);
+        buffer += rep_levels_bytes;
+        data_size -= rep_levels_bytes;
+      }
+      //TODO figure a way to set max_definition_level_ to 0
+      //if the initial value is invalid
```

Comment:
Yes, it shouldn't occur unless we have invalid files or we have a bug in the tree schema resolution. I just left a note because you had code to handle that bit and I removed it.